### PR TITLE
Adds modern js pipeline with latest editor

### PR DESCRIPTION
**Pull request by cybtachyon**
feat(js)

### CODE REVIEW

**Comment by cybtachyon** [comment]
Primary file is js/src/patternkit.jsoneditor.es6.ts

Lib file is js/src/DrupalImageEditor.es6.ts

Everything is else build process / typescript defs / library updates.
Build process is self-documented via package.json and gulpfile.js

File: js/patternkit.jsoneditor.es6.js [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        JSONEditor.defaults.options.collapse = false;
+        JSONEditor.defaults.options.ajax = true;
+        JSONEditor.defaults.editors.drupal_image = DrupalImageEditor;
+        JSONEditor.defaults.resolvers.unshift(function (schema) {
+          if (schema.type === 'string' && schema.format === 'image') {
+            return 'drupal_image';
+          }
+        });
+        // Override how references are resolved.
+        JSONEditor.prototype._loadExternalRefs = function (schema, callback) {
```
Comment (SiliconValet) [inline js/patternkit.jsoneditor.es6.js]: why is this needed?

Comment (cybtachyon) [reply js/patternkit.jsoneditor.es6.js]: This allows us to load refs using URL requests to the patternkit API, which resolves to a file ref.

The API could also hit a REST endpoint URL that returns the schema etc etc.

This allows us to use OOTB JSON Schema as well as namespace-based refs without needing to recursively load all schemas into the editor.
